src/Catalog/SortingCollector.cs

```diff
@@ -10,7 +10,7 @@
 
 namespace NuGet.Services.Metadata.Catalog
 {
-    public abstract class SortingCollector : CommitCollector
+    public abstract class SortingCollector<T> : CommitCollector
```

Comment:
after writing tests i discovered that implementing `IEquatable<T>` isn't enough to make it work--you have to also override `GetHashCode`--so `where T : IEquatable<T>` doesn't have a whole lot of value here.

is there some interface that you know of that forces an override of `GetHashCode`?